prow/cmd/checkconfig/main_test.go

```diff
@@ -357,3 +360,80 @@ func TestOrgRepoUnion(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateUnknownFields(t *testing.T) {
+	testCases := []struct {
+		name        string
+		configBytes []byte
+		config      interface{}
+		expectedErr error
+	}{
+		{
+			name: "valid config",
+			configBytes: []byte(`plugins:
+  kube/kube:
+  - size
+  - config-updater
+config_updater:
+  maps:
+    # Update the plugins configmap whenever plugins.yaml changes
+    kube/plugins.yaml:
+      name: plugins
+size:
+  s: 1`),
+			expectedErr: nil,
+		},
+		{
+			name: "invalid top-level property",
+			configBytes: []byte(`plugins:
+  kube/kube:
+  - size
+  - config-updater
+notconfig_updater:
+  maps:
+    # Update the plugins configmap whenever plugins.yaml changes
+    kube/plugins.yaml:
+      name: plugins
+size:
+  s: 1`),
+			expectedErr: fmt.Errorf("unknown fields present: notconfig_updater"),
+		},
+		{
+			name: "invalid second-level property",
+			configBytes: []byte(`plugins:
+  kube/kube:
+  - size
+  - config-updater
+size:
+  xs: 1
+  s: 5`),
+			expectedErr: fmt.Errorf("unknown fields present: size.xs"),
+		},
+		{
+			name: "invalid array element",
+			configBytes: []byte(`plugins:
+  kube/kube:
+  - size
+  - trigger
+triggers:
+- repos:
+  - kube/kube
+- repoz:
+  - kube/kubez`),
+			expectedErr: fmt.Errorf("unknown fields present: triggers[1].repoz"),
+		},
+	}
```

Comment:
Please add test cases for array and map entry validation.

Reply:
Added test case for [map entry validation](https://github.com/kubernetes/test-infra/pull/11622/files#diff-1bccbd8d60a651295f7c8bb39ac87d4dR434)